CRM/Mailing/BAO/SMSJob.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/**
+ * Job for SMS deliery functions.
+ */
+class CRM_Mailing_BAO_SMSJob extends CRM_Mailing_BAO_MailingJob {
+
+  /**
+   * This is used by CiviMail but will be made redundant by FlexMailer.
+   * @param array $fields
+   *   List of intended recipients.
+   *   Each recipient is an array with keys 'hash', 'contact_id', 'email', etc.
+   * @param $mailing
+   * @param $mailer
+   * @param $job_date
+   * @param $attachments
+   *
+   * @return bool|null
+   * @throws Exception
+   */
+  public function deliverGroup(&$fields, &$mailing, &$mailer, &$job_date, &$attachments) {
+    $count = 0;
+    // dev/core#1768 Get the mail sync interval.
+    $mail_sync_interval = Civi::settings()->get('civimail_sync_interval');
+    $retryGroup = FALSE;
+
+    foreach ($fields as $field) {
+      $contact = civicrm_api3('Contact', 'getsingle', ['id' => $field['contact_id']]);
+
+      $preview = civicrm_api3('Mailing', 'preview', [
+        'id' => $mailing->id,
+        'contact_id' => $field['contact_id'],
+      ])['values'];
+      $mailParams = [
+        'text' => $preview['body_text'],
+        'toName' => $contact['display_name'],
+        'job_id' => $this->id,
+      ];
+      CRM_Utils_Hook::alterMailParams($mailParams, 'civimail');
+      $body = $mailParams['text'];
+      $headers = ['To' => $fields['phone']];
```

Comment:
I'm surprised this worked at all? This should be `$field['phone']`.

Reply:
Opps - yeah that change was a tidy up I made right before commit